Design_Patterns.md

```diff
@@ -0,0 +1,74 @@
+# Design Patterns for Beginners
+
+## Introduction
+Design patterns are solutions to common problems encountered in software design. They provide a 'systematic' yet creative way to design and solve problems. When you're building software, you often face similar challenges, and design patterns offer a way to address these challenges in an efficient manner.

```

Comment:
Perhaps you want to mention that Design Patterns can be applied to any programming languages, because they are a way of thinking instead of a specific algorithm.

Reply:
I have mentioned this in my latest change